src/models/__spec__/component-params.spec.ts

```diff
@@ -1,35 +1,33 @@
-import { ModuleFactory } from '../../factory';
```

Comment:
I think it may be worth looking at turning ComponentParams into an interface (or nothing?). The constructor is rarely used, if at all anymore.

Reply:
Removed and swapped with just an object